### PR TITLE
Migrate lit related web components (lit-element, lit-html) codebase to Lit 2

### PR DESCRIPTION
**Pull request by sync**
Issue:

## What I did

Migrated codebase to Lit 2 (lit-element 3, lit-html 2) (RC)

As you may not be aware, but lit-element (now called lit) has had a fairly big new launch last week with Lit 2.
They have a new website [here](https://lit.dev)

Sadly this new version, in RC at the moment, isn't compatible with what we currently have in storybook so we need to tweak a few things. This is a breaking change and I am pretty sure, once this is merged, storybook will no longer support lit-element 2, or lit-html 1.

## How to test

- Is this testable with Jest or Chromatic screenshots? yes
- Does this need a new example in the kitchen sink apps? no, updated examples
- Does this need an update to the documentation? yes

Would require to update the documentation to mention it will only support Lit 2, I don't see an easy way where both version could end up being supported.

cc @justinfagnani do we have an estimated timeline on when we should expect a non RC version to drop?

### CODE REVIEW

File: addons/docs/src/frameworks/web-components/web-components-properties.test.ts [hunk 1]
```diff
@@ -24,10 +24,6 @@ const runWebComponentsAnalyzer = (inputPath: string) => {
 };
 
 describe('web-components component properties', () => {
-  // we need to mock lit-html and dynamically require custom-elements
-  // because lit-html is distributed as ESM not CJS
-  // https://github.com/Polymer/lit-html/issues/516
-  jest.mock('lit-html', () => {});
```
Comment (sync) [inline addons/docs/src/frameworks/web-components/web-components-properties.test.ts:30]: no longer necessary, see changes made to the main jest.config.js

File: jest.config.js [hunk 2]
```diff
@@ -49,6 +51,7 @@ module.exports = {
     '^.+\\.[jt]sx?$': '<rootDir>/scripts/utils/jest-transform-js.js',
     '^.+\\.mdx$': '@storybook/addon-docs/jest-transform-mdx',
   },
+  transformIgnorePatterns: [`<rootDir>/node_modules/(?!${esModules})`],
```
Comment (sync) [inline jest.config.js:54]: without this jest will end up having issues with those packages

**Comment by sync** [comment]
ran test locally and not seeing any of those errors:

```
Test Suites: 152 passed, 152 total
Tests:       1873 passed, 1873 total
Snapshots:   541 passed, 541 total
Time:        22.215 s, estimated 23 s
Ran all test suites in 8 projects.
```

File: jest.config.js (hunk 2 above)
Comment (merceyz) [reply jest.config.js:54]: This pattern assumes they'll always get hoisted which isn't guaranteed to happen

Comment (sync) [reply jest.config.js:54]: Yes right. Want me to remove rootdir ?

`/node_modules/(?!${esModules})`

**Comment by shilman** [comment]
@sync Thanks so much for your PR! We're actually creating a @storybook/lit over in #14600 to avoid a breaking change and also to name the package more appropriately. Feedback welcome! I'm going to close this for now; we can re-open if needed.

**Pull request closed without merging**